src/main/java/twofive/command/DeleteCommand.java

```diff
@@ -0,0 +1,44 @@
+package twofive.command;
+
+import java.io.IOException;
+
+import twofive.data.TaskList;
+import twofive.exception.InvalidTaskException;
+import twofive.storage.Storage;
+import twofive.task.Task;
+import twofive.ui.Ui;
+
+/**
+ * Deletes a task given its number in a list of tasks when command is executed.
+ */
+public class DeleteCommand extends Command {
+    private int taskNum;
+
+    public DeleteCommand(int taskNum) {
+        this.taskNum = taskNum;
+    }
+
+    /**
+     * Deletes a task from the given list of tasks using the provided task number.
+     *
+     * @param tasks List of tasks to delete the task from.
+     * @param ui UI interacting with user.
+     * @param storage Storage for saving or loading tasks.
+     * @throws InvalidTaskException if task number is invalid
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws InvalidTaskException {
+        if (this.taskNum < 0 || this.taskNum >= tasks.getTasksNum()) {
```

Comment:
Perhaps you could avoid the use of this keyword to reduce unecessary noise
```suggestion
        if (taskNum < 0 || taskNum >= tasks.getTasksNum()) {
```